pkg/sshd/lock.go

```diff
@@ -0,0 +1,67 @@
+package sshd
+
+import (
+	"fmt"
+	"sync"
+)
+
+// RepositoryLock interface that allows the creation of a lock associated
+// with a repository name to avoid simultaneous git operations
+type RepositoryLock interface {
+	// Lock aquires a lock for a repository
+	Lock(repoName string) error
+	// Unlock releases the lock for a repository or returns
+	// an error if the specified name doesn't exist.
+	Unlock(repoName string) error
+}
+
+// NewRepositoryLock returns a new instance of a key mutex.
+func NewRepositoryLock() RepositoryLock {
+	return &repoLock{
+		mutexMap: make(map[string]bool),
+	}
+}
+
+type repoLock struct {
+	sync.RWMutex
+	mutexMap map[string]bool
+}
+
+// Lock aquires a lock associated with the specified name
+func (rl *repoLock) Lock(repoName string) error {
+	rl.RWMutex.Lock()
```

Comment:
no because that produces a data race (I tried your suggestion using `go test -race -cover lock.go lock_test.go`)